lambda_functions/analyzer/binary_info.py

```diff
@@ -71,13 +72,9 @@ def __enter__(self):
         return self
 
     def __exit__(self, exception_type, exception_value, traceback):
-        """Remove the downloaded binary from local disk."""
-        # In Lambda, "os.remove" does not actually remove the file as expected.
-        # Thus, we first truncate the file to set its size to 0 before removing it.
-        if os.path.isfile(self.download_path):
-            with open(self.download_path, 'wb') as file:
-                file.truncate()
-            os.remove(self.download_path)
+        """Shred the downloaded binary and delete it from disk."""
+        # Note: This runs even during exception handling (it is the "with" context).
+        subprocess.check_call(['shred', '--remove', self.download_path])
```

Comment:
will we know/handle if the `shred` operation fails?

Reply:
The `check_call` will raise a runtime exception if the exit code is non-zero, which will halt the Lambda function immediately. There's not much else we could do to handle it - any error would likely mean the filesystem is unresponsive